vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -1431,84 +1400,71 @@ public void fetchCurrentUser(@NotNull ModelCallback<User> callback, @Nullable St
      * @param query         Query string for hitting the search endpoint
      * @param refinementMap Used to refine lists (generally for search) with sorts and filters
      * @param fieldFilter   The string of fields to include in the response (highly recommended!)
-     *                      {@link SearchRefinementBuilder}
+     *                      {@link RequestRefinementBuilder}
+     * @param caller        The {@link Caller} for the expected response type
      * @see <a href="https://developer.vimeo.com/api/spec#common-parameters">Vimeo API Field Filter Docs</a>
      */
     @Nullable
-    public Call<Object> fetchContent(@NotNull String uri, CacheControl cacheControl,
-                                     @NotNull ModelCallback callback, @Nullable String query,
-                                     @Nullable Map<String, String> refinementMap,
-                                     @Nullable String fieldFilter) {
+    public <DataType_T> Call<DataType_T> getContent(@NotNull String uri,
```

Comment:
From a bigger picture perspective, a better solution would be to remove the Caller from the networking library entirely. The need to dynamically switch out the call is an artifact of how streams is architected in our app, but it makes calling the networking library more confusing than it needs to be for someone that knows they want to call a specific endpoint and get a specific typed result. My goal in this refactor was to find a minimal-impact way to introduce type-safety so I was trying to keep a close analogy between fetchContent and getContent with the introduction of the Caller objects. I'd like to revisit this architecture in the near future with an eye to revising or removing how this works, but it was the best idea I could come up with for solving the type-safety problem with minimal scope.